src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -330,12 +335,27 @@ private boolean checkImmutableIndices(Object request, ActionListener listener) {
         }
         
         if(request instanceof IndexRequest) {
-            if(complianceConfig != null && complianceConfig.isIndexImmutable(request)) {
+            if (isIndexImmutable(request)) {
                 ((IndexRequest) request).opType(OpType.CREATE);
             }
         }
         
         return false;
     }
 
-}
\ No newline at end of file
+    private boolean isIndexImmutable(Object request) {
+        if (complianceConfig == null || !complianceConfig.isEnabled()) {
+            return false;
+        }
+
+        final Set<String> immutableIndicesPatterns = complianceConfig.getImmutableIndicesPatterns();
+        if (immutableIndicesPatterns.isEmpty()) {
```

Comment:
Should this be part of the check on line 198? You can check that `immutableIndicesPatterns` is not empty once for `BulkShardRequest` and assume that it is not empty here or pass it to `checkImmutableIndices()`.